src/python/ensembl/io/genomio/assembly/status.py

```diff
@@ -0,0 +1,473 @@
+    incorrect parameterization.
+
+    Args:
+        input_cores: input core(s) list file name.
+        input_accessions: input accession (s) list file name.
+        db_host: host server name
+        db_port: host serer port
+
+    Returns:
+        User input file used in assembly status querying
+    """
+    user_query_file: Path
+
+    # Check for required input in the form of cores/accessions
+    if input_cores is None and input_accessions is None:
+        logging.critical(
+            "Missing required input: '--input_cores' (core db names) OR '--input_accns' (INSDC accessions)."
+        )
+        sys.exit()
+    elif input_cores and input_accessions:
+        logging.critical("Detected '--input_cores' AND '--input_accns'. Please provide just one such option.")
+        sys.exit()
+    # Input core names centered run
+    elif input_cores and input_accessions is None:
+        user_query_file = input_cores
+        logging.info(f"Performing assembly status report using core db list file: {user_query_file}")
+        if db_host is None or db_port is None:
+            logging.critical(
+                "User must specify both arguments '--host' and '--port' when providing core database names."
+            )
+            sys.exit()
+    # Accession centered run
+    elif input_cores is None and input_accessions:
+        user_query_file = input_accessions
+        logging.info(f"Performing assembly status report using INSDC accession list file: {user_query_file}")
+
+    return user_query_file
+
+
+def resolve_query_type(
+    query_list: list, host_server: str, host_port: str, input_cores: str, input_accessions: str
+) -> Union[Tuple[Dict, str]]:
+    """Function to indentify the kind of querys being passed by user,
+    then extract the queries (core names or accesisons) and store each with appropriate identifier.
+
+    Args:
+        query_list: List of user defined queries either core names, or accessions
+        input_cores: arg parse param '--input_cores'
+        input_accessions: arg parse param '--input_accns'
+
+    Returns:
+        User queries stored as indentifier[(core db name | UniqueID#)] : accession
+    """
+
+    query_accessions: Dict = {}
+    query_type: str = ""
+
+    if input_cores and input_accessions is None:
+        server_details = f"mysql://ensro@{host_server}:{host_port}/"  ## Requires some more dev !!
+        query_accessions = fetch_asm_accn(query_list, server_details)
+        query_type = "CoreDB"
+    elif input_cores is None and input_accessions:
+        query_count = 1
+        query_type = "Accession"
+        for accession in query_list:
+            match = re.match(r"(GC[AF])_([0-9]{3})([0-9]{3})([0-9]{3})\.?([0-9]+)", accession)
+            if not match:
+                raise UnsupportedFormatError(f"Could not recognize GCA accession format: {accession}")
+            query_name = f"Query_#{query_count}"
```

Comment:
Nitpicking: the header says `Accession`, so why is the query name a number and not the accession?

Reply:
Literally to avoid redundancy of information have the same accession in two different locations on the report TSV. Im not married to this way though, happy to change if you feel its silly

Reply:
Using accession now instead of arbitrary "query+number"